### PR TITLE
Convert the remaining classic classes to native classes

### PR DESCRIPTION
**Pull request by Windvis**
This builds on #83  since that has the updated linting config, but I'll rebase once reviewed.

### CODE REVIEW

File: app/adapters/application.js [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
 
-    const origAjax = this._super;
     const args = arguments;
     const sleep = (time) => new Promise( (resolve/*,reject*/) => setTimeout(() => resolve(true), time));
     const retry = (livesSpent) => {
       return new Promise((resolve, reject) => {
         sleep(250 * livesSpent).then(() => {
           if (livesSpent < maxRetries )
-            origAjax.apply(this,args).then( (res) => resolve(res), () => retry(livesSpent + 1).then(
+            super.ajax(...args).then((res) => resolve(res), () => retry(livesSpent + 1).then(
```
Comment (Windvis) [inline app/adapters/application.js:14]: I need to double check this tomorrow.

File: app/utils/promise-proxy-object.js [hunk 2]
```diff
@@ -4,4 +4,4 @@ import PromiseProxyMixin from '@ember/object/promise-proxy-mixin';
 // see https://emberjs.com/api/ember/3.7/classes/PromiseProxyMixin for
 // info on how this is used.
 
-export default ObjectProxy.extend(PromiseProxyMixin);
+export default class PromiseProxyObject extends ObjectProxy.extend(PromiseProxyMixin) {}
```
Comment (Windvis) [inline app/utils/promise-proxy-object.js:7]: I think this util can be removed, the code that used this seems to be removed as well.

Comment (nvdk) [reply app/utils/promise-proxy-object.js:7]: should be ok to remove indeed :+1:

**Review by nvdk — CHANGES_REQUESTED** [review]
looks ok, but indeed the ajax overwrite is now broken.

File: app/adapters/application.js (hunk 1 above)
Comment (Windvis) [reply app/adapters/application.js]: Fixed the issue and rewrote the method while I was at it. I think it's more readable now (and easier to extract into a shareable util).

File: .eslintrc.js [hunk 3]
```diff
@@ -26,8 +26,6 @@ module.exports = {
     browser: true,
   },
   rules: {
-    'ember/avoid-leaking-state-in-ember-objects': 'warn',
```
Comment (Windvis) [inline .eslintrc.js:29]: These overrides are now no longer needed :tada:

**Pull request merged**